src/service/github.oauth.service.ts

```diff
@@ -0,0 +1,71 @@
+import * as express from "express";
+import { Server } from "http";
+import fetch from "node-fetch";
+import { URLSearchParams } from "url";
+import { state } from "../state";
+
+export class GitHubOAuthService {
+  public app: express.Express;
+  public server: Server;
+
+  constructor(public port: number) {
+    this.app = express();
+    this.app.use(express.json(), express.urlencoded({ extended: false }));
+  }
+
+  public async StartProcess() {
+    this.server = this.app.listen(this.port);
+    this.app.get("/callback", async (req, res) => {
+      const params = new URLSearchParams(
+        await (await this.getToken(req.param("code"))).text()
+      );
+
+      res.send("<script>window.close();</script>");
```

Comment:
in Firefox new tab didnt close ? Can you look into it.
As we need to close the tabs in all browsers.

Reply:
I wasn't able to get this to work because of limitations from modern browsers, but I have made a simple screen informing the user that they may close the tab.

![image](https://user-images.githubusercontent.com/25834068/60195777-ca458180-984c-11e9-8b9f-c3f6fd58fd4e.png)

Reply:
sounds good!